### PR TITLE
Get and list collection info

### LINKED ISSUE #396: Get collection info

### LINKED ISSUE #397: List collections

### PR DESCRIPTION
**Pull request by sanderpick**
Closes #396 
Closes #397

### CODE REVIEW

File: api/client/client.go [hunk 1]
```diff
@@ -343,27 +343,82 @@ func (c *Client) DeleteCollection(ctx context.Context, dbID thread.ID, name stri
...
 			Path:   index.Path,
 			Unique: index.Unique,
 		}
 	}
-	return indexes, nil
+	return indexes
+}
+
+// GetCollectionIndexes returns an existing collection's indexes.
+// Deprecated: Use GetCollectionInfo instead.
```
Comment (sanderpick) [inline api/client/client.go:385]: GetCollectionInfo has the indexes as well

File: db/index.go [hunk 2]
```diff
@@ -47,11 +47,16 @@ type Index struct {
...
-	indexes := make([]Index, len(c.indexes))
+	if len(c.indexes) == 0 {
+		return nil
+	}
+	indexes := make([]Index, len(c.indexes)-1)
 	var i int
 	for _, index := range c.indexes {
-		indexes[i] = index
-		i++
+		if index.Path != idFieldName {
```
Comment (sanderpick) [inline db/index.go:56]: If we include the index on `_id`, then it has to be filtered by the user if they're going to use these indexes on a _new_ collection, i.e., you can't create a custom index on that field.

Comment (carsonfarmer) [reply db/index.go:56]: Ah good 'catch'

File: api/pb/javascript/package.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
   "dependencies": {
     "@improbable-eng/grpc-web": "^0.12.0",
     "@types/google-protobuf": "^3.7.2",
-    "google-protobuf": "^3.12.1"
+    "google-protobuf": "^3.12.2"
```
Comment (carsonfarmer) [inline api/pb/javascript/package.json:20]: 👍

**Comment by carsonfarmer** [comment]
@carsonfarmer  and @andrewxhill this requires some API updates in @textile/threads-client.

**Pull request merged**